org.omg.sysml/src/org/omg/sysml/util/repository/APIModelDelta.java

```diff
@@ -0,0 +1,226 @@
+		List<DataVersion> delta = new LinkedList<>();
+		
+		for (Element addition: getAdditions()) {
+			Data data = new Data();
+			for (String field: addition.keySet()) {
+				Object value = addition.get(field);
+				data.put(field, value);
+			}
+			UUID id = UUID.fromString(addition.get(Identified.SERIALIZED_NAME_AT_ID).toString());
+			delta.add(new DataVersion().payload(data).identity(new DataIdentity().atId(id)));
+		}
+		
+		for (var change: changes.entrySet()) {
+			delta.add(new DataVersion().payload(change.getValue().toData()).identity(new DataIdentity().atId(change.getKey())));
+		}
+		
+		for (UUID deletion: getDeletions()) {
+			//set to null for deletion
+			delta.add(new DataVersion().payload(null).identity(new DataIdentity().atId(deletion)));
+		}
+		
+		return delta;
+	}
+	
+	/**
+	 * JSON string of the delta
+	 * 
+	 * @param gson JSON parser to use
+	 * @return JSON string
+	 */
+	public String toJson(Gson gson) {
+		return gson.toJson(toTrasferableDelta());
+	}
+	
+	/**
+	 * JSON string of the delta
+	 * 
+	 * @param gson JSON parser to use
+	 * @return JSON string
+	 */
+	public JsonElement toJsonTree(Gson gson) {
+		return gson.toJsonTree(toTrasferableDelta());
+	}
+	
+	/**
+	 * Creates an {@link APIModelDelta} between two {@link APIModel}s
+	 * 
+	 * @param thisModel model to compare with baseline
+	 * @param baseline model to use as a baseline
+	 * 
+	 * @return {@link APIModelDelta} of the two models
+	 */
+	public static APIModelDelta create(APIModel thisModel, APIModel baseline) {
+		if (thisModel == null) thisModel = APIModel.createEmpty();
+		if (baseline == null) baseline = APIModel.createEmpty();
+		
+		Set<Element> additions = new HashSet<>();
+		Set<UUID> deletions = new HashSet<>();
+		Map<UUID, ChangedElement> changes = new HashMap<>();
+		
+		Map<UUID, Element> elementsFromThis = thisModel.getModelElements();
+		
+		for (UUID id: elementsFromThis.keySet()) {
+			Element baselineElement = baseline.getElement(id);
+			if (baselineElement == null) {
+				additions.add(thisModel.getElement(id));
+			} else {
+				Element thisElement = thisModel.getElement(id);
+				ChangedElement change = createElementDelta(thisElement, baselineElement);
+				if (change != null) {
+					changes.put(id, change);
+				}
+			}
+		}
+		
+		Map<UUID, Element> elementsFromBaseline = baseline.getModelElements();
+		
+		for (UUID id: elementsFromBaseline.keySet()) {
+			Element element = thisModel.getElement(id);
+			if (element == null) {
+				deletions.add(id);
+			}
+		}
+		
+		return new APIModelDelta(additions, changes, deletions);
+	}
+	
+	/**
+	 * Creates a {@link Data} object which contains changed fields and their new values
+	 * 
+	 * @param thisElement element to compare with baseline
+	 * @param baselineElement element to use as a baseline
+	 * @return changed fields and their new values wrapped in a {@link Data} object.
+	 */
+	public static ChangedElement createElementDelta(Element thisElement, Element baselineElement) {

```

Comment:
The name `thisElement` is strange here, as `this` is an entire delta. One option would be to rename it (consistently with other uses) to simply element (remove the prefix). Other alternatives might be to prefix it as `current` or `head`.